### DISCUSSION TITLE
"Truly" Global Parameters in MOOSE input file for parsed functions

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#25259] [Q&A General] [answered]
Is there a way to define truly global parameters in the input file? Say I have a parameter that is included in several locations e.g used in parsed functions, is it possible to define it in one location?

In the Pasred Functions block, you need to define "symbol_names" and "symbol_values" below it with the value. In some other blocks that use parsed functions (e.g ParsedGenerateSideset, you need to define "constant_names" and "constant_expressions".

Instead of needing to specify for each block, is there a way to specify a 1 to 1 "replacement"? That is, wherever the input file sees "x0" it automatically replaces it with some value you define? This would be useful for other blocks too, say, within the executioner, you define the dtmin and dtmax at the top of the file as some kind of parameter, say "dtmin_val = 0.2", so you can write:

"dtmin = dtmin_val" in the executioner block and the input file automatically replaces it with 0.2? Even though, originally, the executioner has no "constant_values"/"constant_expressions" option.

I tried using the "GlobalParams" system but that didn't work. 

The purpose for this is that I want to write a bash script to loop over several values of a parameter and run the simulation each time, but that's kinda hard to do when you have to go looking through the whole file for the various places you define the parameter.

**Comment by GiudGiud** [comment, marked as answer]
You can define parameters at the top of the input file then refer to them in each parameter:
```
abc = 3

[Auxkernels]
  [p]
    type = ParsedAux
    expression = 'a + efg'
    symbol_names = 'efg'
    symbol_values = '${abc}'
[][]
```
for example

**Reply by richmondodufisan** [reply]
This is what I was looking for. Thank you.

**Reply by richmondodufisan** [reply]
@GiudGiud quick question, do parsed functions also work? I'm trying to define the simulation time at the top of the file as well. Following syntax in https://mooseframework.inl.gov/application_usage/input_syntax.html, I have:

`t_val = ${fparse 2.2*${period}*${tp} }`

gives the error

`invalid character '}' - did you leave a field value blank after a previous '='?`

using quotes i.e

`t_val = ${fparse '2.2*${period}*${tp}' }`

gives the same error.

**Reply by GiudGiud** [reply]
use this
`${fparse 2.2*period*tp }`

**Reply by richmondodufisan** [reply]
perfect. thanks!
